docs/guides/testing.md

```diff
@@ -51,13 +56,19 @@ import { act } from 'react-dom/test-utils'
 const storeResetFns = new Set<() => void>()
 
 // when creating a store, we get its initial state, create a reset function and add it in the set
-export const create = <S>(createState: StateCreator<S>) => {
+const createInternalFn = <S>(createState: StateCreator<S>) => {
   const store = actualCreate(createState)
   const initialState = store.getState()
   storeResetFns.add(() => store.setState(initialState, true))
   return store
 }
 
+export const create = <S>(createState: StateCreator<S>) => {
+  return typeof createState === 'function'
+    ? createInternalFn(createState)
+    : createInternalFn
```

Comment:
From previous PR:

I don't know exactly how it works, and I'm on my phone, so can't check it, but shouldn't it be:

```suggestion
    : createInternalFn()
```

or something like:

```suggestion
    : createInternalFn(defaultStateCreator)
```

?

Right now the effective type of `create` is different, depending on the inputs:
`StateCreator => State` or `Not<Function> => StateCreator => State`.

On the second thought, can `StateCreator` _not_ be a function? Because TypeScript should give an error if someone passes something different, so a runtime type check is unnecessary then.